### DISCUSSION TITLE
Custom template

### DISCUSSION THREAD

**Discussion by JamesGai** [hakimio/ngx-ui-tour#199] [Q&A]
Can I please have an example of how to implement a custom guide template?

**Comment by hakimio** [comment]
Have you checked the documentation site: https://hakimio.github.io/ngx-ui-tour/md-menu/Misc#custom-template

**Reply by JamesGai** [reply]
Yes, the documentation site only contains one HTML example, but I'm a bit confused. Do I also need to implement variables `step` and other `tourService` functions using import `TourService` in a separate .ts file?

Also in my custom HTML template, do I also need to change the selector `<tour-step-template> ... </tour-step-template>` as shown below to my custom name?
```
<tour-step-template>
    <ng-template let-step="step">
        ...
     </ng-template>
</tour-step-template>
```

**Reply by JamesGai** [reply]
Providing a full example would be great, thanks!

**Comment by hakimio** [comment]
> Do I also need to implement variables step and other tourService functions using import TourService in a separate .ts file?

The `step` variable is provided for you from the template. `TourService` you have to inject in your component.
